### PR TITLE
feat: informative error message

### PR DESCRIPTION
**Pull request by VoVAllen**
Signed-off-by: Jinjing.Zhou <allenzhou@tensorchord.ai>

### CODE REVIEW

**Review by VoVAllen — COMMENTED** [review]
Found error trace is not informative in most cases. Finally decide not to add it. Only debug mode will print the full error stacktrace.

Other changes:
- For buildkit error, it will print the full error trace for user easier to understand the detailed error
- Starlark list -> String slice will panic before if element is not string inside the list. Now check before conversion

File: pkg/lang/frontend/starlark/starlark_utils/stringslice.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+package starlarkutils
+
+import (
+	"github.com/cockroachdb/errors"
+
+	"go.starlark.net/starlark"
+)
+
+func ToStringSlice(v *starlark.List) ([]string, error) {
```
Comment (gaocegege) [inline pkg/lang/frontend/starlark/starlark_utils/stringslice.go]: Could we move it to pkg/util?

File: pkg/builder/builder.go [hunk 2]
```diff
@@ -280,7 +290,8 @@ func (b generalBuilder) build(ctx context.Context, pw progresswriter.Writer) err
 				}
 				_, err := b.Client.Build(ctx, solveOpt, "envd", b.BuildFunc(), pw.Status())
 				if err != nil {
-					err = errors.Wrap(err, "failed to solve LLB")
+					err = errors.Wrap(&BuildkitdErr{err: err}, "Buildkit error")
+					logrus.Errorf("%+v", err)
```
Comment (terrytangyuan) [inline pkg/builder/builder.go:294]: Use %w to format errors

File: cmd/envd/main.go [hunk 3]
```diff
@@ -44,18 +45,20 @@ func handleErr(err error) {
...
-	rootCause := errors.Cause(err)
 	var evalErr *starlark.EvalError
 	var syntaxErr *syntax.Error
+	var resolveErr resolve.ErrorList
 	if ok := errors.As(err, &evalErr); ok {
 		fmt.Fprintln(os.Stderr, evalErr.Backtrace())
 	} else if ok := errors.As(err, &syntaxErr); ok {
 		fmt.Fprintln(os.Stderr, syntaxErr)
+	} else if ok := errors.As(err, &resolveErr); ok {
+		fmt.Fprintf(os.Stderr, "%+v\n", resolveErr)
```
Comment (terrytangyuan) [inline cmd/envd/main.go:59]: Use %w to format errors

File: pkg/builder/builder.go (hunk 2 above)
Comment (VoVAllen) [reply pkg/builder/builder.go:294]: This is intentional. %+v is a feature of `cockroachdb/errors` to print the full error messages including stacktrace.

File: cmd/envd/main.go (hunk 3 above)
Comment (VoVAllen) [reply cmd/envd/main.go:59]: ditto

File: pkg/lang/frontend/starlark/starlark_utils/stringslice.go (hunk 1 above)
Comment (VoVAllen) [reply pkg/lang/frontend/starlark/starlark_utils/stringslice.go]: Moved

**Comment by gaocegege** [comment]
LGTM

/cc @kemingy

**Review by kemingy — APPROVED** [review]
/lgtm

**Pull request merged**